ETL/Python ETL/MatchHistoryETL.py

```diff
@@ -0,0 +1,117 @@
+                               "AY": "AwayTeamYellowCards",
+                               "HR": "HomeTeamRedCards",
+                               "AR": "AwayTeamRedCards",
+                               "B365H": "B365HomeTeam",
+                               "B365D": "B365Draw",
+                               "B365A": "B365AwayTeam",
+                               "MaxH": "MarketMaxHomeTeam",
+                               "MaxD": "MarketMaxDraw",
+                               "MaxA": "MarketMaxAwayTeam",
+                               "AvgH": "MarketAvgHomeTeam",
+                               "AvgD": "MarketAvgDraw",
+                               "AvgA": "MarketAvgAwayTeam",
+                               "B365>2.5": "B365Over2.5Goals",
+                               "B365<2.5": "B365Under2.5Goals",
+                               "Max>2.5": "MarketMaxOver2.5Goals",
+                               "Max<2.5": "MarketMaxUnder2.5Goals",
+                               "Avg>2.5": "MarketAvgOver2.5Goals",
+                               "Avg<2.5": "MarketAvgUnder2.5Goals"},
+                   inplace = True)
+
+# Add season column
+website_frame.insert(0, "Season", constants.CURRENT_SEASON_TAG)
+
+# Add MatchWeek value
+website_frame.insert(1, "MatchWeek", constants.DEFAULT_MATCHWEEK)
+
+# Add Points columns
+conditions = [
+     website_frame["FullTimeResult"] == 'H',
+     website_frame["FullTimeResult"] == 'D',
+     website_frame["FullTimeResult"] == 'A'
+]
+
+home_points = [ 3, 1, 0]
+away_points = [ 0, 1, 3]
+
+website_frame["HomeTeamPoints"] = np.select(conditions, home_points)
+website_frame["AwayTeamPoints"] = np.select(conditions, away_points)
+
+# Stablish a connection to Database data source and fetch all matches stored from current season
+try:
+    connection = psycopg2.connect(
+        host = constants.DB_SERVER,
+        port = constants.DB_PORT,
+        user = constants.DB_USER,
+        password = constants.DB_PASSWORD,
+        database = constants.DB_NAME
+    )
+except psycopg2.Error as e:
+    print (f'Can not connect to the postgress database "{constants.DB_NAME}". Make sure database server is running')
+    print (e)
+else:
+    print (f'Connection to database "{constants.DB_NAME}" stablished. Listening at port {constants.DB_PORT}')
+
+season_query = f"SELECT * FROM public.match_history WHERE \"Season\" = '{constants.CURRENT_SEASON_TAG}'"
+
+cursor = connection.cursor()
+cursor.execute(season_query)
+matches_in_db = cursor.fetchall()
+
+# Copy cursor into a dataframe
+postgres_frame = pd.DataFrame(data = matches_in_db, columns = website_frame.columns)
+
+# !! Getting error --> ValueError: Can only compare identically-labeled (both index and columns) DataFrame objects
+# This does not work as our dataframes have identical lables, but different indexes as website_frame is always likely to have
+# more entries that what we have persisted in DataBase
+new_entries = website_frame.compare(postgres_frame)
```

Comment:
Is this the best way to know what are new matches to insert into database?

Reply:
Oh ok that sounds like a good idea.
Ok so I understand the staging table part, but .. how could I run a script that tries to insert only new fresh data into a permanent table? do I need triggers or something? 
Also, this idea reminds me to bring to the table DBT
DBT is a tool I know a little bit and definitely would like to learn, do you think this is a perfect use case to add DBT to the team?

Reply:
so one way to insert matches is by using the [postgres upsert ](https://www.postgresqltutorial.com/postgresql-tutorial/postgresql-upsert/) 

another way you can implement it is using subqueries, where you use the primary key to join incoming data and existing data, and only insert new rows into the main fact table

Reply:
About dbt, I was thinking of introducing that to a later stage. But, dbt is now a days the transformation layer in the . the [modern data stack](https://www.getdbt.com/blog/future-of-the-modern-data-stack)